scripts/functions-deploy-tests/tests.ts

```diff
@@ -260,6 +263,83 @@ describe("firebase deploy", function (this) {
     }
   });
 
+  it("skips duplicate deploys functions with runtime options", async () => {
+    const opts: Opts = {
+      v1Opts: {
+        memory: "128MB",
+        maxInstances: 42,
+        timeoutSeconds: 42,
+      },
+      v2Opts: {
+        memory: "128MiB",
+        maxInstances: 42,
+        timeoutSeconds: 42,
+        cpu: 2,
+        concurrency: 42,
+      },
+      v1TqOpts: {
+        retryConfig: {
+          maxAttempts: 42,
+          maxRetrySeconds: 42,
+          maxBackoffSeconds: 42,
+          maxDoublings: 42,
+          minBackoffSeconds: 42,
+        },
+        rateLimits: {
+          maxDispatchesPerSecond: 42,
+          maxConcurrentDispatches: 42,
+        },
+      },
+      v2TqOpts: {
+        retryConfig: {
+          maxAttempts: 42,
+          maxRetrySeconds: 42,
+          maxBackoffSeconds: 42,
+          maxDoublings: 42,
+          minBackoffSeconds: 42,
+        },
+        rateLimits: {
+          maxDispatchesPerSecond: 42,
+          maxConcurrentDispatches: 42,
+        },
+      },
+      v1IdpOpts: {
+        blockingOptions: {
+          idToken: true,
+          refreshToken: true,
+          accessToken: false,
+        },
+      },
+      v2IdpOpts: {
+        idToken: true,
+        refreshToken: true,
+        accessToken: true,
+      },
+      v1ScheduleOpts: {
+        retryCount: 3,
+        minBackoffDuration: "42s",
+        maxRetryDuration: "42s",
+        maxDoublings: 42,
+        maxBackoffDuration: "42s",
+      },
+      v2ScheduleOpts: {
+        schedule: "every 30 minutes",
+        retryCount: 3,
+        minBackoffSeconds: 42,
+        maxRetrySeconds: 42,
+        maxDoublings: 42,
+        maxBackoffSeconds: 42,
+      },
+    };
+
+    previews.skipdeployingnoopfunctions = true;
```

Comment:
q: do we plan on removing this preview flag? If so, we should put a TODO here to yank it

Reply:
Updated!